src/Native/Unix/System.Security.Cryptography.Native/pal_rsa.cpp

```diff
@@ -67,12 +94,31 @@ extern "C" int32_t CryptoNative_RsaGenerateKeyEx(RSA* rsa, int32_t bits, BIGNUM*
 extern "C" int32_t
 CryptoNative_RsaSign(int32_t type, const uint8_t* m, int32_t mlen, uint8_t* sigret, int32_t* siglen, RSA* rsa)
 {
-    if (!siglen)
+    if (siglen == nullptr)
     {
         assert(false);
         return 0;
     }
 
+    *siglen = 0;
+
+    if (HasNoPrivateKey(rsa))
+    {
+        ERR_PUT_error(ERR_LIB_RSA, RSA_F_RSA_SIGN, RSA_R_VALUE_MISSING, __FILE__, __LINE__);
+        return 0;
+    }
+
+    // Shared pointer to the metadata about the message digest algorithm
+    const EVP_MD* digest = EVP_get_digestbynid(type);
+
+    // If the digest itself isn't known then RSA_R_UNKNOWN_ALGORITHM_TYPE will get reported, but
+    // we have to check that the digest size matches what we expect.
+    if (digest != nullptr && mlen != EVP_MD_size(digest))
```

Comment:
No, see the comment.  There is already handling for not recognizing the algorithm, I'm just providing the missing error handling for the size being wrong.